### PR TITLE
#2336 BE: implemented endpoint for getting of exercises containing sp…

### PR DESCRIPTION
**Pull request by mironovandrei**
https://github.com/Brain-up/brn/issues/2336
[BE Admin] Design and implement api for searching exercises by word

Implemented endpoint "/api/exercises/byWord?word=..." which returns list of exercises containing specified word

### CODE REVIEW

File: src/main/kotlin/com/epam/brn/service/ExerciseService.kt [hunk 1]
```diff
@@ -154,6 +155,11 @@ class ExerciseService(
         return subGroupExercises.map { it.toDto() }
     }
 
+    fun findExerciseByWord(word: String): List<ExerciseWithWordsResponse> {
```
Comment (ElenaSpb) [inline src/main/kotlin/com/epam/brn/service/ExerciseService.kt:158]: findExerciseByWord ->
findExercisesByWord

File: src/main/kotlin/com/epam/brn/dto/response/ExerciseWithWordsResponse.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+data class ExerciseWithWordsResponse(
+    val id: Long?,
+    val name: String,
+    val active: Boolean = true,
+    val changedBy: String? = "InitialDataLoader",
+    val changedWhen: LocalDateTime = LocalDateTime.now(ZoneOffset.UTC),
+    val playWordsCount: Int? = 1,
+    val wordsColumns: Int? = 3,
+    val words: List<String>,
```
Comment (ElenaSpb) [inline src/main/kotlin/com/epam/brn/dto/response/ExerciseWithWordsResponse.kt:14]: also it should return a Resource.id
as our use case is to 
1.find exercises with wrong words
2.fix them (here we need resource.id to make call to back for edit)
so maybe it should be a map<resourceId, word>
?

File: src/main/kotlin/com/epam/brn/service/ExerciseService.kt (hunk 1 above)
Comment (mironovandrei) [reply src/main/kotlin/com/epam/brn/service/ExerciseService.kt]: renamed

File: src/main/kotlin/com/epam/brn/dto/response/ExerciseWithWordsResponse.kt (hunk 2 above)
Comment (mironovandrei) [reply src/main/kotlin/com/epam/brn/dto/response/ExerciseWithWordsResponse.kt]: ok
added Resource.id as a map <resourseId, word>

**Review by ElenaSpb — APPROVED** [review]

**Pull request merged**